url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/client/TradeOfferPanel.java

```diff
@@ -0,0 +1,820 @@
+                        w = buttonX + buttonW + ShadowedBox.SHADOW_SIZE + 2;
+                } else {
+                    // Buttons below giveLab2, offerSquares
+                    final int buttonY = top + 8 + (2 * squaresHeight) + lineH;
+
+                    sendBut.setBounds(inset, buttonY, buttonW, buttonH);
+                    clearBut.setBounds(inset + 5 + buttonW, buttonY, buttonW, buttonH);
+                    cancelBut.setBounds(inset + (2 * (5 + buttonW)), buttonY, buttonW, buttonH);
+                }
+
+                if (counterCompactMode)
+                {
+                    // No balloon point, so top h/8 of its bounding box is empty
+                    balloon.setBounds(0, -h/8, w, h - (h/16));
+                } else {
+                    balloon.setBounds(0, 0, w, h);
+                }
+                offerBox.setBounds(0, top + 18 + squaresHeight, w, squaresHeight + 16 + lineH);
+
+                // If offerBox height calculation changes, please update OFFER_COUNTER_HEIGHT.
+            }
+            else
+            {
+                balloon.setBalloonPoint(true);
+
+                int lineH = ColorSquareLarger.HEIGHT_L;
+                int giveW = fm.stringWidth("I Give: ") + 2;
+
+                toWhom1.setBounds(inset, top, w - 20, 14);
+                toWhom2.setBounds(inset, top + 14, w - 20, 14);
+                giveLab.setBounds(inset, top + 32, giveW, lineH);
+                getLab.setBounds(inset, top + 32 + lineH, giveW, lineH);
+                squares.setLocation(inset + giveW, top + 32);
+                squares.doLayout();
+                
+                if (offered)
+                {
+                    int squaresHeight = squares.getBounds().height + 8;
+                    int buttonY = top + 32 + squaresHeight;
+                    acceptBut.setBounds(inset, buttonY, buttonW, buttonH);
+                    rejectBut.setBounds(inset + 5 + buttonW, buttonY, buttonW, buttonH);
+                    offerBut.setBounds(inset + (2 * (5 + buttonW)), buttonY, buttonW, buttonH);
+                }
+
+                balloon.setBounds(0, 0, w, h);
+
+                // If rejectBut height calculation changes, please update OFFER_HEIGHT.
+                // If change in the height dfference of "offered" buttons showing/not showing,
+                // please update OFFER_BUTTONS_HEIGHT.
+            }
+        }
+
+        /**
+         * Respond to button-related user input
+         *
+         * @param e Input event and source
+         */
+        public void actionPerformed(ActionEvent e)
+        {
+            try {
+            String target = e.getActionCommand();
+
+            if (target == OFFER)
+            {
+                setCounterOfferVisible(true);
+            }
+            else if (target == CLEAR)
+            {
+                offerSquares.setValues(zero, zero);
+            }
+            else if (target == SEND)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.